test/com/google/enterprise/adaptor/opentext/OpentextAdaptorTest.java

```diff
@@ -3053,6 +3537,137 @@ public GroupPrincipal pushGroupDefinitions(
+  }
+
+  @Test
+  public void testGroupsNullMembers() throws InterruptedException {
+    class MemberServiceMockError extends MemberServiceMock {
+      @Override
+      public MemberSearchResults getSearchResults(PageHandle pageHandle) {
+        return new MemberSearchResultsMock((PageHandleMock) pageHandle) {
+          @Override
+          public List<Member> getMembers() {
+            return null;
+          }
+        };
+      }
+    }
+    SoapFactoryMock soapFactory = new SoapFactoryMock();
+    soapFactory.memberServiceMock = new MemberServiceMockError();
+    soapFactory.memberServiceMock.addMember(
+        getMember(2000, "group 1", "Group"));
+    OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+    AdaptorContext context = ProxyAdaptorContext.getInstance();
+    Config config = initConfig(adaptor, context);
+    adaptor.init(context);
+    OpentextAdaptor.Groups groups =
+        adaptor.new Groups(soapFactory.newMemberService());
+    groups.addGroups();
+    assertEquals(ImmutableMap.of(), groups.getGroupDefinitions());
+    assertEquals(true, groups.isPartialResults());
+  }
+
+  @Test
+  public void testGroupsPublicAccessNullHandle()
+      throws InterruptedException {
+    class MemberServiceMockError extends MemberServiceMock {
+      @Override
+      public PageHandle searchForMembers(MemberSearchOptions options) {
+        return null;
+      }
+    }
+    SoapFactoryMock soapFactory = new SoapFactoryMock();
+    soapFactory.memberServiceMock = new MemberServiceMockError();
+    OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+    AdaptorContext context = ProxyAdaptorContext.getInstance();
+    Config config = initConfig(adaptor, context);
+    adaptor.init(context);
+    OpentextAdaptor.Groups groups =
+        adaptor.new Groups(soapFactory.newMemberService());
+    groups.addPublicAccessGroup("[Public Access]", LOCAL_NAMESPACE);
+    assertEquals(ImmutableMap.of(), groups.getGroupDefinitions());
+    assertEquals(true, groups.isPartialResults());
+  }
+
+  @Test
+  public void testGroupsPublicAccessNullSearchResults()
+      throws InterruptedException {
+    class MemberServiceMockError extends MemberServiceMock {
+      @Override
+      public MemberSearchResults getSearchResults(PageHandle pageHandle) {
+        return null;
+      }
+    }
+    SoapFactoryMock soapFactory = new SoapFactoryMock();
+    soapFactory.memberServiceMock = new MemberServiceMockError();
+    soapFactory.memberServiceMock.addMember(getMember(2000, "user 1", "User"));
+    OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+    AdaptorContext context = ProxyAdaptorContext.getInstance();
+    Config config = initConfig(adaptor, context);
+    adaptor.init(context);
+    OpentextAdaptor.Groups groups =
+        adaptor.new Groups(soapFactory.newMemberService());
+    groups.addPublicAccessGroup("[Public Access]", LOCAL_NAMESPACE);
+    assertEquals(1, groups.getGroupDefinitions().size());
```

Comment:
Maybe just something like this (might need type parameter tweaks), here and below:

assertEquals(
    ImmutableMap.of(newGroupPrincipal("[Public Access]"), ImmutableList.of()),
    groups.getGroupDefinitions());

Reply:
Not sure why I said "here and below", but I guess I meant "here and above". You changed lines 3632ff, but not this one.